app/helpers/spree/products_helper_decorator.rb

```diff
@@ -0,0 +1,10 @@
+Spree::ProductsHelper.module_eval do
+
+  def variant_full_price(variant)
+    product = variant.product
+    unless product.variants.active(current_currency).all? { |v| v.price == product.price && v.sale_price == product.price }
+      Spree::Money.new(variant.sale_price || variant.price, currency: current_currency).to_html
```

Comment:
@rishabhguptarishi sale price will be 0.0 by default and not nil hence this will not work.